mesa/visualization/video_viz.py

```diff
@@ -0,0 +1,184 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+import mesa
+from mesa.visualization.matplotlib_renderer import (
+    MatplotlibRenderer,
+    MeasureRendererMatplotlib,
+    SpaceRenderMatplotlib,
+)
+
+
+def make_space_component(
+    agent_portrayal: Callable | None = None,
+    propertylayer_portrayal: dict | None = None,
+    post_process: Callable | None = None,
+    **space_drawing_kwargs,
+):
+    """Create a Matplotlib-based space visualization component.
+
+    Args:
+        agent_portrayal: Function to portray agents.
+        propertylayer_portrayal: Dictionary of PropertyLayer portrayal specifications
+        post_process : a callable that will be called with the Axes instance. Allows for fine tuning plots (e.g., control ticks)
+        backend: The backend to use for rendering the space. Can be "matplotlib" or "altair".
+        space_drawing_kwargs : additional keyword arguments to be passed on to the underlying space drawer function. See
+                               the functions for drawing the various spaces for further details.
+
+    ``agent_portrayal`` is called with an agent and should return a dict. Valid fields in this dict are "color",
+    "size", "marker", and "zorder". Other field are ignored and will result in a user warning.
+
+
+    Returns:
+        SpaceRenderMatplotlib: A component for rendering the space.
+    """
+    if agent_portrayal is None:
+
+        def agent_portrayal(a):
+            return {}
+
+    return SpaceRenderMatplotlib(
+        agent_portrayal,
+        propertylayer_portrayal,
+        post_process=post_process,
+        **space_drawing_kwargs,
+    )
+
+
+def make_measure_component(
+    measure: Callable,
+    **kwargs,
+):
+    """Create a plotting function for a specified measure.
+
+    Args:
+        measure (str | dict[str, str] | list[str] | tuple[str]): Measure(s) to plot.
+        kwargs: Additional keyword arguments to pass to the MeasureRendererMatplotlib constructor.
+
+    Returns:
+        MeasureRendererMatplotlib: A component for rendering the measure.
+    """
+    return MeasureRendererMatplotlib(
+        measure,
+        **kwargs,
+    )
+
+
+class VideoViz:
+    """Create high-quality video recordings of model simulations."""
+
+    def __init__(
+        self,
+        model: mesa.Model,
+        components: Sequence[MatplotlibRenderer],
+        *,
+        title: str | None = None,
+        figsize: tuple[float, float] | None = None,
+        grid: tuple[int, int] | None = None,
+    ):
+        """Initialize video visualization configuration.
+
+        Args:
+            model: The model to simulate and record
+            components: Sequence of component objects defining what to visualize
+            title: Optional title for the video
+            figsize: Optional figure size in inches (width, height)
+            grid: Optional (rows, cols) for custom layout. Auto-calculated if None.
+        """
+        # Check if FFmpeg is available
```

Comment:
This is perhaps for users the most tricky part. I agree on the error handling here.  I would suggest also adding this to the module level docstring.